agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/processors/ExampleExporter.java

```diff
@@ -0,0 +1,342 @@
+        if(config!=null) {
+            SpanData updatedSpan = processOtherActions(config,span,includeFlag,excludeFlag,spanName,existingSpanAttributes);
+            return processInsertActions(config,updatedSpan,includeFlag,excludeFlag,spanName,existingSpanAttributes);
+        }
+
+        return span;
+    }
+
+    private SpanData processInsertActions(SpanProcessorConfig config, SpanData span, Boolean includeFlag, Boolean excludeFlag, String spanName,
+                                          ReadableAttributes existingSpanAttributes) {
+
+        final Builder insertBuilder = Attributes.newBuilder();
+
+        if(config.insertActions!=null && config.insertActions.size()>0) {
+            if(includeFlag==null) {
+                includeFlag = checkIncludes(config.include,existingSpanAttributes,spanName);
+            }
+            if(excludeFlag==null) {
+                excludeFlag=checkExcludes(config.exclude,existingSpanAttributes,spanName);
+            }
+            if(includeFlag && !excludeFlag) {
+
+                for(SpanProcessorAction actionObj:config.insertActions) {
+                    if(actionObj.key!=null && actionObj.value!=null) {
+                        insertBuilder.setAttribute(actionObj.key, actionObj.value);
+                    }
+                }
+                span.getAttributes().forEach(new KeyValueConsumer<AttributeValue>() {
+                    public void consume(String key, AttributeValue value) {
+                        insertBuilder.setAttribute(key, value);
+                    }
+                });
+                return new MySpanData(span, insertBuilder.build());
+            }
+        }
+
+        return span;
+    }
+
+    private SpanData processOtherActions(final SpanProcessorConfig config, SpanData span, Boolean includeFlag, Boolean excludeFlag, String spanName,
+                                     final ReadableAttributes existingSpanAttributes) {
+        final Builder builder = Attributes.newBuilder();
+        if(config.otherActions != null && config.otherActions.size() > 0 ) {
+            if(includeFlag==null) {
+                includeFlag = checkIncludes(config.include,existingSpanAttributes,spanName);
+            }
+            if(excludeFlag==null) {
+                excludeFlag=checkExcludes(config.exclude,existingSpanAttributes,spanName);
+            }
+            if(includeFlag && !excludeFlag) {
+                // loop over existing attributes
+                span.getAttributes().forEach(new KeyValueConsumer<AttributeValue>() {
+
+                    @Override
+                    public void consume(String key, AttributeValue value) {
+                        // flag to check if a attribute is updated
+                        boolean updatedFlag=false;
+                        for(SpanProcessorAction actionObj:config.otherActions) {
+                            if(actionObj.key!=null && actionObj.action!=null && actionObj.key.equals(key)) {
```

Comment:
can be simplified with optimized model
```suggestion
                            if(actionObj.key.equals(key)) {
```